app/src/main/java/com/gophillygo/app/data/models/Filter.java

```diff
@@ -0,0 +1,149 @@
+    }
+
+    private Filter(Parcel in) {
+        boolean[] fields = in.createBooleanArray();
+        this.nature = fields[0];
+        this.exercise = fields[1];
+        this.educational = fields[2];
+        this.been = fields[3];
+        this.wantToGo = fields[4];
+        this.notInterested = fields[5];
+        this.liked = fields[6];
+        this.accessible = fields[7];
+    }
+
+    public int count() {
+        boolean[] fields = {nature, exercise, educational, been, wantToGo, notInterested, liked, accessible};
+        int selectedCount = 0;
+        for (boolean field : fields) {
+            if (field) {
+                selectedCount++;
+            }
+        }
+        return selectedCount;
+    }
+
+    public boolean matches(DestinationInfo info) {
+        boolean categoryMatches = categoryMatches(info.getDestination().getCategories());
+        boolean flagMatches = flagMatches(info.getFlag());
+        boolean accessibleMatches = accessibleMatches(info.getDestination().isAccessible());
+
+        return categoryMatches && flagMatches && accessibleMatches;
+    }
+
+    public boolean matches(EventInfo info) {
+        boolean categoryMatches = categoryMatches(info.getDestinationCategories());
+        boolean flagMatches = flagMatches(info.getFlag());
+        boolean accessibleMatches = accessibleMatches(info.getEvent().isAccessible());
+
+        return categoryMatches && flagMatches && accessibleMatches;
+    }
+
+    private boolean categoryMatches(List<String> destCategories) {
+        if (destCategories == null) return false;
+
+        boolean categoryMatches = categories().isEmpty();
+        for (String category : categories()) {
+            if (destCategories.contains(category)) {
+                categoryMatches = true;
+            }
+        }
+        return categoryMatches;
+    }
+
+    private boolean flagMatches(AttractionFlag flag) {
+        boolean flagMatches = flags().isEmpty();
+        for (AttractionFlag.Option option : flags()) {
+            if (flag.getOption() == option) {
+                flagMatches = true;
+            }
+        }
+        return flagMatches;
+    }
+
+    private boolean accessibleMatches(boolean isAccessible) {
+        boolean accessibleMatches = true;
+        if (accessible && !isAccessible) {
+            accessibleMatches = false;
+        }
+        return accessibleMatches;
+    }
+
+    private List<String> categories() {
+        List<String> categories = new ArrayList<>();
+        if (nature) {
+            categories.add("Nature");
+        }
+        if (exercise) {
+            categories.add("Exercise");
+        }
+        if (educational) {
+            categories.add("Educational");
+        }
+        return categories;
+
+    }
+
+    private List<AttractionFlag.Option> flags() {
+        List<AttractionFlag.Option> flags = new ArrayList<>();
+        if (been) {
+            flags.add(AttractionFlag.Option.Been);
```

Comment:
Prefer getting all the options from `AttractionFlag.Option` and iterating over them, rather than listing them again here, which is brittle.